students/lee_deitesfeld/lesson04/mailroom.py

```diff
@@ -17,113 +18,74 @@
 
-dic_donors = {'donor_db':
-                [
-                {"name": "Bob Jones",
-                 "donations": [63.00, 30.00]
-                },
-                {"name": "Sue Smith",
-                 "donations": [50.00]
-                },
-                {"name": "Joe Grimes",
-                 "donations": [102.00, 45.00, 70.00]
-                },
-                {"name": "Andrea Funk",
-                 "donations": [21.00, 21.00]
-                },
-                {"name": "Mickey Mouse",
-                 "donations": [500.00, 250.00, 100.00]
-                }
-                ]}
+dic_donors = {"Bob Jones": [63.00, 30.00],
+             "Sue Smith": [50.00],
+             "Joe Grimes": [102.00, 45.00, 70.00],
+             "Andrea Funk": [21.00, 21.00],
+             "Mickey Mouse": [500.00, 250.00, 100.00]}
 
 def menu_selection(prompt, main_dispatch):
+    """Pulls up main menu"""
     while True:
         response = input(prompt)
         if main_dispatch[response]() == "exit menu":
             break
 
 def sub_menu():
+    """Pulls up sub menu"""
     menu_selection(sub_prompt, sub_dispatch)
 
 def quit():
+    """Quits current menu"""
     print("Quitting this menu")
     return "exit menu"
 
 def report():
-    '''Draws up report of names and donations'''
-
-    #calculates sum of donations and adds new key-value pair to donor_db dict
-    for x in dic_donors['donor_db']:
-        ttl_amt = sum(x['donations'])
-        x['sum'] = ttl_amt
-
-    #calculates number of donations and adds new key-value pair to donor_db dict
-    for x in dic_donors['donor_db']:
-        numofdonations = len(x['donations'])
-        x['count'] = numofdonations
-
-    #calculates average of donations and adds new key-value pair to donor_db dict
-    for x in dic_donors['donor_db']:
-        average = sum(x['donations'])/len(x['donations'])
-        str_average = f'{average:.2f}' #limits average to two decimal points
-        x['average'] = str_average
-
-    #sorts list of donors on total amount donated
-    for x in dic_donors['donor_db']:
-        int_ttl_amt = int(x['sum']) #convert ttlamt to integer so sort will work on entire digit (not just first)
-        x['sum'] = int_ttl_amt
-    dic_donors['donor_db'].sort(key=itemgetter('sum'), reverse = True) #sort on key='sum'
-    for x in dic_donors['donor_db']:
-        ttl_amt_2dec = '%.2f' % x['sum'] #limits total to two decimal points
-        x['sum'] = ttl_amt_2dec
-
-    #removes history of donations, since it's not in final report
-    for x in dic_donors['donor_db']:
-        del x['donations']
-
-    #formats final report
+    """Creates a report of donors and donation amounts"""
+    report = []
+    for x in dic_donors:
+        ttl_amt = sum(dic_donors[x])
+        numofdonations = len(dic_donors[x])
+        average = ttl_amt/numofdonations
+        report.append((x, ttl_amt, numofdonations, average))
+        report.sort(key=itemgetter(1), reverse = True)
```

Comment:
it looks like you're sorting for every item in your list? I think you want this outside of the for loop (de-dent)